libraries/cmake/source/libgcrypt/CMakeLists.txt

```diff
@@ -0,0 +1,223 @@
+    "${library_root}/cipher/gostr3411-94.c"
+    "${library_root}/cipher/crc.c"
+    "${library_root}/cipher/stribog.c"
+    "${library_root}/cipher/md5.c"
+    "${library_root}/cipher/md4.c"
+    "${library_root}/cipher/rmd160.c"
+    "${library_root}/cipher/sha256.c"
+    "${library_root}/cipher/sha512.c"
+    "${library_root}/cipher/keccak.c"
+    "${library_root}/cipher/whirlpool.c"
+    "${library_root}/cipher/tiger.c"
+    "${library_root}/cipher/blake2.c"
+    "${library_root}/cipher/sha1.c"
+    "${library_root}/cipher/scrypt.c"
+    "${library_root}/cipher/gost28147.c"
+    "${library_root}/compat/compat.c"
+    "${library_root}/mpi/mpi-add.c"
+    "${library_root}/mpi/mpi-gcd.c"
+    "${library_root}/mpi/mpi-mul.c"
+    "${library_root}/mpi/mpi-cmp.c"
+    "${library_root}/mpi/mpi-inline.c"
+    "${library_root}/mpi/mpi-bit.c"
+    "${library_root}/mpi/mpi-div.c"
+    "${library_root}/mpi/mpi-mod.c"
+    "${library_root}/mpi/mpi-inv.c"
+    "${library_root}/mpi/mpi-pow.c"
+    "${library_root}/mpi/mpi-scan.c"
+    "${library_root}/mpi/mpicoder.c"
+    "${library_root}/mpi/mpih-div.c"
+    "${library_root}/mpi/mpi-mpow.c"
+    "${library_root}/mpi/mpiutil.c"
+    "${library_root}/mpi/mpih-mul.c"
+    "${library_root}/mpi/ec-ed25519.c"
+    "${library_root}/mpi/ec.c"
+    "${library_root}/random/random.c"
+    "${library_root}/random/random-csprng.c"
+    "${library_root}/random/random-system.c"
+    "${library_root}/random/rndjent.c"
+    "${library_root}/random/rndhw.c"
+    "${library_root}/random/random-drbg.c"
+    "${library_root}/random/rndlinux.c"
+    "${library_root}/src/hwfeatures.c"
+    "${library_root}/src/sexp.c"
+    "${library_root}/src/visibility.c"
+    "${library_root}/src/global.c"
+    "${library_root}/src/hmac256.c"
+    "${library_root}/src/stdmem.c"
+    "${library_root}/src/misc.c"
+    "${library_root}/src/missing-string.c"
+    "${library_root}/src/secmem.c"
+    "${library_root}/src/fips.c"
+    "${library_root}/src/dumpsexp.c"
+    "${library_root}/src/hwf-x86.c"
+    "${library_root}/src/context.c"
+    "${library_root}/src/mpicalc.c"
+
+    "${CMAKE_CURRENT_SOURCE_DIR}/generated/mpi/mpih-add1.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/generated/mpi/mpih-sub1.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/generated/mpi/mpih-mul1.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/generated/mpi/mpih-mul2.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/generated/mpi/mpih-mul3.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/generated/mpi/mpih-lshift.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/generated/mpi/mpih-rshift.c"
+  )
+
+  # Some of these files don't play nice with clang; let's compile them
+  # manually with gcc and then link in the object files
+  #
+  # The avx and avx2 files are NOT included here so that we are compatible
+  # with more processors. Note that the configuration flags to disable them
+  # were not enough! Make sure all the AVX/AVX2 options are off when re-generating
+  # a new config file!
+  set(asm_source_list
```

Comment:
This is why we are still using gcc; some of those .S files do not play nice with clang. This could be maybe fixed with `-no-integrated-as` and some additional parameters!

Reply:
Just as reminder, we would still have to depend on the gcc package (only when not using our custom toolchain though), but at least the compilation would be consistent, using the same frontend.

Reply:
I'm testing the build right now using clang and only installing binutils, seems to work, but I'll let my CI finish to see if there's anything wrong.

Reply:
Works fine.